lib/oban/telemetry.ex

```diff
@@ -158,4 +158,19 @@ defmodule Oban.Telemetry do
       |> Jason.encode!()
     end)
   end
+
+  @doc """
+  Measure the time it took to run a function and publishes a telemetry
+  """
+  @spec span(name :: atom(), fun :: (() -> term()), metadata :: map()) :: term()
+  def span(name, fun, metadata \\ %{}) do
```

Comment:
Reading through this now and seeing how it is used in practice I don't think `span` is the correct name. According to the telemetry.span proposal there would be `start` and `stop` events for each span call. That doesn't fit well with wrapping the execution of a single function so I believe this needs a different name. Something like `wrap` or even `execute`?

Reply:
Hi @sorentwo! The way this is implemented could actually work well for a span, but the behaviour within this `span` function needs to be extended.  The proposal in beam-telemetry/telemetry#57 defines three events that make up a span: `:start`, `:stop`, and `:exception`.

Reply:
Thanks for linking to the proposal @mcrumm. I've followed the span discussion a bit, and that's why I suggested using `span` inside of Oban. My thought here was that I'm not sure these are "spans" because only a single function is being executed. However, seeing your support for it being a `span` and thinking about it as a way to maintain consistency, I like it.